src/index.js

```diff
@@ -278,14 +281,12 @@ export default function undoable (reducer, rawConfig = {}) {
           return config.history
         }
 
-        if (config.filter && typeof config.filter === 'function') {
-          if (!config.filter(action, res, history.present)) {
-            debug('filter prevented action, not storing it')
-            debugEnd()
-            return {
-              ...history,
-              present: res
-            }
+        if (typeof config.filter === 'function' && !config.filter(action, res, history.present)) {
+          debug('filter prevented action, not storing it')
+          debugEnd()
+          return {
+            ...history,
+            present: res
           }
```

Comment:
first `if(config.filter)` can be removed here, typeof will never fail. Or was this done because typeof might be slower? And not everybody uses the filter? @omnidan 

And we can also combined the 2 if's here.